backend/infrahub/core/query/diff.py

```diff
@@ -604,7 +604,7 @@ async def query_init(self, db: InfrahubDatabase, **kwargs):
                     WHERE r.from <= $to_time AND (r.to IS NULL or r.to >= $to_time)
                     AND r.branch IN $branch_names
                 )
-                AND ID(inner_p) <> ID(prop)
+                AND [ID(inner_p), type(inner_diff_rel)] <> [ID(prop), type(r_prop)]
```

Comment:
this conditional was meant to prevent us from traversing from node A to node B back to node A, but we actually _want_ to do this if node A is the peer AND the owner/source of node B